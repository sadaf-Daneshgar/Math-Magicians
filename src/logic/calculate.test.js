import calculate from './calculate';

describe('calculate function', () => {
  it('should match the snapshot for AC button', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, 'AC');
    expect(result).toMatchSnapshot();
  });

  it('should match the snapshot for number buttons', () => {
    const result1 = calculate({ total: '10', next: '5', operation: '+' }, '2');
    const result2 = calculate({ total: '10', next: null, operation: null }, '0');

    expect(result1).toMatchSnapshot();
    expect(result2).toMatchSnapshot();
  });
});
