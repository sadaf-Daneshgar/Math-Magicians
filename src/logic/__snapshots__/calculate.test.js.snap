// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate function should match the snapshot for AC button 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;

exports[`calculate function should match the snapshot for number buttons 1`] = `
Object {
  "next": "52",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate function should match the snapshot for number buttons 2`] = `
Object {
  "next": "0",
  "total": null,
}
`;
